prototype2/client/src/engine/control/camera.rs

```diff
@@ -20,36 +20,30 @@ impl System {
   }
 }
 
-#[allow(unused_imports, unused_variables)]
 impl specs::System<engine::Delta> for System {
   fn run(&mut self, arg: specs::RunArg, _: engine::Delta) {
-    use specs::Join;
     use itertools::Itertools;
 
-    let (mut camera_events,
-         mut camera_pos,
-         own_entity,
-         physical,
-         synchronized,
-         entities,
-         window) = arg.fetch(|w| {
+    let (mut camera_events, mut camera_pos, menu_state, window) = arg.fetch(|w| {
       (w.write_resource::<Vec<CameraMoveEvent>>(),
        w.write_resource::<CameraPos>(),
-       w.read_resource::<Option<OwnEntity>>(),
-       w.read::<PhysicalAspect>(),
-       w.read::<SynchronizedAspect>(),
-       w.entities(),
+       w.read_resource::<menu::MenuState>(),
        w.read_resource::<glutin::Window>())
     });
 
-    // Move the mouse back to the middle of the window
-    let (wx, wy) = window.get_position().unwrap();
-    let (ox, oy) = window.get_outer_size().unwrap();
-    let (middle_x, middle_y) = ((wx + ox as i32 / 2), (wy + oy as i32 / 2));
-    window.set_cursor_position(middle_x, middle_y).unwrap();
+    if !menu_state.is_open() {
+      // Move the mouse back to the middle of the window
+      let (wx, wy) = window.get_position().unwrap();
+      let (ox, oy) = window.get_outer_size().unwrap();
+      let (middle_x, middle_y) = ((wx + ox as i32 / 2), (wy + oy as i32 / 2));
```

Comment:
This is now moved into event_distribution, where it hasn't been extracted because the middle_x value is needed for computing the camera event. Went the lazy route for lack of a good function name.